text/0022-response-context.md

```diff
@@ -0,0 +1,144 @@
+
+# Options Considered
+
+## Option 1
+
+Adding a `Response` interface directly in the [Event payload](https://develop.sentry.dev/sdk/event-payloads/).
+
+```json
+{
+  "response": {
+    // ..
+  }
+}
+```
+
+## Option 2 (Chosen)
+
+Adding a `Response` interface in the [Contexts interface](https://develop.sentry.dev/sdk/event-payloads/contexts/).
+
+```json
+{
+    "contexts": {
+        "response": {
+            "type": "response"
+            // ...
+  }
+}
+```
+
+## Option 3
+
+Expand the `Request` interface adding the missing fields.
+
+Data scrubbing should consider response headers when scrubbing.
+
+If we do that, the `Request` docs should be ammended that it contains the `Response` data as well otherwise it's semantically wrong.
+
+# Proposal
+
+The proposal is the Option 2, adding a `Response` interface in the [Contexts interface](https://develop.sentry.dev/sdk/event-payloads/contexts/).
+
+By doing the Option 2, we can keep the `Request` interface as it is and we don't need to change the data scrubbing rules for the `Request` field.
+
+Adding it as part of the `Contexts`, we get a lot for free such as retained arbitrary fields and back compatibility.
+
+```json
+{
+  "contexts": {
+    "response": {
+      "type": "response",
+      "cookies": "PHPSESSID=298zf09hf012fh2; csrftoken=u32t4o3tb3gg43; _gat=1;",
+      "headers": {
+        "content-type": "text/html"
+      },
+      "inferred_content_type": "text/html",
+      "status_code": 500,
+      "is_redirect": false,
+      "body_size": 1000,
+      "arbitrary_field": "arbitrary" // arbitrary and retained fields for backwards compatibility when adding new fields
+    }
+  }
+}
+```
+
+* `type`: `response` as `String`.
+* `cookies`: Can be given unparsed as `String`, as `Dictionary`, or as a `List of Tuples`.
+* `headers`: A `Dictionary` of submitted headers, this requires a special treatment in the data scrubbing rules.
+* `status_code`: The HTTP status code, `Integer`.
+* `is_redirect`: A `Boolean` indicating if the response was a redirect.
+* `body_size`: A `Number` (absolute/positive) indicating the size of the response body in bytes.
+* `inferred_content_type`: A `String` indicating the inferred content type of the response.
+
+The `url`, `method`, `query_string`, `fragment`, `env` fields are not part of the `Response` interface and they should be set under the `Request` field, even if inferred from the HTTP response in case you don't have control over the HTTP Request object.
+
+The `data` field won't be added to the `Response` interface, a phase 2 of this RFC will propose add Request and Response bodies are sent as attachments.
+
+## Must have
+
+A tag should be created for `Request#url` and `Response#status_code` fields, people should be able to search for events with a specific `url` and/or `status_code`, also to alert on them.
+
+# Drawbacks
+
+The `Response` interface is PII sensitive, so we need to be careful about how we scrub it.
```

Comment:
Would it make sense to define the scrubbing behavior in the field list above?

Reply:
The fields that have PII are similar to the request one, so I guess it's fine.
I added one sentence about this tho, mainly `cookies` and `headers`.